js/bundle.go

```diff
@@ -252,22 +252,21 @@ func (b *Bundle) getExports(logger logrus.FieldLogger, rt *goja.Runtime, options
 }
 
 // Instantiate creates a new runtime from this bundle.
-func (b *Bundle) Instantiate(logger logrus.FieldLogger, vuID uint64) (bi *BundleInstance, instErr error) {
-	// TODO: actually use a real context here, so that the instantiation can be killed
-	// Placeholder for a real context.
-	ctxPtr := new(context.Context)
-
+func (b *Bundle) Instantiate(
+	logger logrus.FieldLogger, vuID uint64, vuImpl *moduleVUImpl,
+) (bi *BundleInstance, instErr error) {
 	// Instantiate the bundle into a new VM using a bound init context. This uses a context with a
 	// runtime, but no state, to allow module-provided types to function within the init context.
-	rt := goja.New()
-	init := newBoundInitContext(b.BaseInitContext, ctxPtr, rt)
-	if err := b.instantiate(logger, rt, init, vuID); err != nil {
+	vuImpl.runtime = goja.New()
```

Comment:
Could we directly inject now a vuImpl with an initialized runtime so we can maintain the initialization in one central place? (e.g. `runner.newVU`)

Reply:
in a followign PR  :P  There are in practice a lot more changes, I just really needed moduleVUImpl to be part of VU and everything else was gravy